app/src/main/java/com/example/pix/home/fragments/HomeFragment.java

```diff
@@ -41,29 +59,71 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         mSpotifyAppRemote = LoginActivity.getmSpotifyAppRemote();
 
         // When we click on the edit profile Toolbar button, replace this screen with a ProfileFragment
-        (view.findViewById(R.id.home_profile_icon)).setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                getActivity().getSupportFragmentManager().beginTransaction()
-                        .setCustomAnimations(R.anim.slide_in_up, R.anim.slide_out_up)
-                        .addToBackStack("stack")
-                        .replace(R.id.home_profile, new ProfileFragment())
-                        .commit();
-            }
-        });
+        (view.findViewById(R.id.home_profile_icon)).setOnClickListener(view12 -> getActivity().getSupportFragmentManager().beginTransaction()
+                .setCustomAnimations(R.anim.slide_in_up, R.anim.slide_out_up)
+                .addToBackStack("stack")
+                .replace(R.id.home_profile, new ProfileFragment())
+                .commit());
 
         PagerTabStrip pagerTabStrip = view.findViewById(R.id.pager_header);
         pagerTabStrip.setDrawFullUnderline(false);
 
-        SearchView svChats = view.findViewById(R.id.search_user);
+        Button svChats = view.findViewById(R.id.search_user);
+
+        // ALMOST ALL OF THIS CODE IS REPEATED FROM COMPOSE, SO LOOK TO REDUCE CODE REPETITION
```

Comment:
`// TODO: ALMOST ALL OF THIS CODE IS REPEATED FROM COMPOSE, SO LOOK TO REDUCE CODE REPETITION`

Reply:
And yes why can't you make the home fragment the same as compose fragment?

Reply:
I decided to not do it for now, since there is one difference (They use different SearchAdapter constructors), and it feels weird to make a method that creates a pop-up since I would need to pass in all the parameters that I would need, like the Activity and Context. I can definitely look into it later though.